### PR TITLE
Introduce tests (part 2)

### PR DESCRIPTION
**Pull request by tyxchen**
Introduces tests for `shared/errors`, `shared/hooks/*`, and `shared/form/*`.

---

Blocked by: none
Blocks: #26

### CODE REVIEW

**Comment by ktrieu** [comment]
Looks pretty much fine, though I'm unsure about throwing an error on an invalid error map. What I've seen with `react-18n` is it displays the key as a fallback instead of throwing an exception and crashing the app.

**Comment by tyxchen** [comment]
I figure that once we get all the testing done, such crashes will be caught in integration testing. So I'd rather throw the error than silently fail.

**Comment by ktrieu** [comment]
Alright, sounds fair to me.

File: src/shared/__tests__/errors.unit.test.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+  it("rejects adding additional errors to nested levels", () => {
+    try {
+      ERRORS.__TEST.__ASDFUHFAI = "?";
+      expect("Error was not thrown by ErrorsProxy").toBe(false);
+    } catch (e) {
+      expect(e.message.includes("Error was not thrown by ErrorsProxy")).toBe(false);
+    }
+  });
+
+  describe("USER.REQUIRED", () => {
```
Comment (ktrieu) [inline src/shared/__tests__/errors.unit.test.ts:86]: We'll probably end up using the required form in more places than the user form, so we should move it to the top-level.

Comment (tyxchen) [reply src/shared/__tests__/errors.unit.test.ts:86]: The way I structured this test suite (and all my test suites) is that top-level test groups all correspond to a module export. We then test functionality related to that export within the test group. If a method or part of an export requires more granular testing (e.g. here with `USER.REQUIRED`), we declare a nested test group. This way, the organization of test suites follow the organization of their corresponding modules.

Comment (ktrieu) [reply src/shared/__tests__/errors.unit.test.ts:86]: I probably worded that weird. What I meant is that in the error map, we should move `USER.REQUIRED` into just `REQUIRED`, seeing as the text is pretty generic and doesn't reference the user form at all.

Comment (tyxchen) [reply src/shared/__tests__/errors.unit.test.ts]: Ahh, I see what you mean. Moved and deprecated in 02c0bea.

**Pull request merged**